src/controllers/plugins/Plugin.hpp

```diff
@@ -0,0 +1,158 @@
+#pragma once
+#ifdef CHATTERINO_HAVE_PLUGINS
+#    include "Application.hpp"
+#    include "controllers/commands/CommandController.hpp"
+
+#    include <magic_enum.hpp>
+#    include <QDir>
+#    include <QJsonArray>
+#    include <QJsonObject>
+#    include <QString>
+#    include <semver/semver.hpp>
+
+#    include <set>
+#    include <vector>
+
+struct lua_State;
+
+namespace chatterino {
+
+struct PluginMeta {
+    // required fields
+    QString name;
+    QString description;
+    QString authors;
+    QString license;
+    semver::version version;
+
+    // optional
+    QString homepage;
+    std::vector<QString> tags;
+
+    bool valid{};
+    std::vector<QString> invalidWhy;
```

Comment:
Maybe have a vector of strings in `errors` and a function `isValid()` that returns `errors.empty()` instead?